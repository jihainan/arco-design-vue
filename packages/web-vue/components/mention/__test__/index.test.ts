import { mount } from '@vue/test-utils';
import Mention from '../index';

describe('Mention', () => {
  test('mention correctly', async () => {
    const wrapper = mount(Mention, {
      props: {
        data: ['Bytedance', 'Bytedesign', 'Bytenumner'],
      },
    });
    const input = wrapper.find('input');
    await input.trigger('focusin');
    await input.setValue('@');

    expect(
      wrapper.findComponent({ name: 'DropdownPanel' }).html()
    ).toMatchSnapshot();
  });

  test('should select value', async () => {
    const wrapper = mount(Mention, {
      props: {
        data: ['Bytedance', 'Bytedesign', 'Bytenumner'],
      },
    });
    const input = wrapper.find('input');
    await input.trigger('focusin');
    await input.setValue('@');
    const dropdown = wrapper.findComponent({ name: 'DropdownPanel' });
    await input.trigger('keydown', { code: 'ArrowDown' });
    expect(dropdown.find('.arco-dropdown-option-active').text()).toBe(
      'Bytedesign'
    );
    await input.trigger('keydown', { code: 'ArrowUp' });
    expect(dropdown.find('.arco-dropdown-option-active').text()).toBe(
      'Bytedance'
    );
    await input.trigger('keydown', { code: 'Enter' });
    expect(wrapper.emitted('change')?.[1]).toEqual(['@Bytedance']);
  });
});
