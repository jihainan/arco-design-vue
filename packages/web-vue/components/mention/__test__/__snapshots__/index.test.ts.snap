// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mention mention correctly 1`] = `
"<div class=\\"arco-dropdown\\">
  <div class=\\"arco-spin\\" style=\\"width: 100%; display: block;\\">
    <!--v-if-->
    <div class=\\"arco-dropdown-list-wrapper\\">
      <ul class=\\"arco-dropdown-list\\">
        <li class=\\"arco-dropdown-option arco-dropdown-option-active\\">
          <!--v-if-->Bytedance
          <!--v-if-->
        </li>
        <li class=\\"arco-dropdown-option\\">
          <!--v-if-->Bytedesign
          <!--v-if-->
        </li>
        <li class=\\"arco-dropdown-option\\">
          <!--v-if-->Bytenumner
          <!--v-if-->
        </li>
      </ul>
    </div>
    <!--v-if-->
    <!---->
  </div>
</div>"
`;
